sdk/eventhub/azure-eventhubs/azure/eventhub/producer.py

```diff
@@ -229,18 +228,18 @@ def send(self, event_data, partition_key=None, timeout=None):
             wrapper_event_data = event_data
         else:
             if isinstance(event_data, EventDataBatch):  # The partition_key in the param will be omitted.
-                if partition_key and not (partition_key == event_data._partition_key):  # pylint: disable=protected-access
+                if partition_key and not partition_key == event_data._partition_key:  # pylint: disable=protected-access
```

Comment:
make it `partition_key != event_data._partition_key`?

Reply:
Changed